crates/mdb/src/error.rs

```diff
@@ -1,21 +1,21 @@
-use std::{
-    ffi::CStr,
-    fmt::{Debug, Display, Formatter},
-};
-
-use libc::c_char;
+use std::ffi::{c_char, CStr};
+use std::fmt::{Debug, Display, Formatter};
+use std::marker::PhantomData;
 
 unsafe fn pchar_to_string(p_value: *const c_char) -> String {
     assert!(!p_value.is_null());
     let value = String::from(CStr::from_ptr(p_value).to_str().unwrap());
     value
 }
 
-enum OwnedOrBorrowedError {
-    Owned(*mut mdb_sys::mdb_error_t),
-    Borrowed(*const mdb_sys::mdb_error_t),
+pub struct Error {
+    ptr: *mut mdb_sys::mdb_error_t,
+}
+
+pub(crate) struct BorrowedError<'a> {
+    ptr: *const mdb_sys::mdb_error_t,
+    _marker: PhantomData<&'a mdb_sys::mdb_error_t>,
```

Comment:
Do you happen to know if `mdb_sys::mdb_error_t` is necessary or if `()` would suffice since we are only trying to capture the lifetime?

Reply:
I'm not actually sure, but I think being verbose here is nice :)